src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubLeaseFactory.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    // EventProcessorHost requires a container name that the multiple host instances use to coordinate with each other and create leases. 
+    // An event hub is described by an (namespace name, event hub path name). The mapping from the event hub to the container name
+    // must be a one-to-one injective function such that:
+    // 1. it's deterministic: when we scale out ot N separate instances, they must all compute the same container name for the same event hub. 
+    // 2. it's collision free: different event hubs shouldn't share the same container. So we can't just use a hash. 
+    //  Note that the event hub name is much more expressive than the storage container name, so we can't do a stateless collision-free transform. 
+    // Hence we use a lookaside storage table to save the transform. 
+    internal class EventHubLeaseFactory
+    {
+        // Name of Azure table that maps "eventHub" --> container name. 
+        private const string TableName = "AzureFunctionsEventHubTable";
+             
+        // Get container name. Lease name that multiple eventhub instances use to coordinate with each other. 
+        public static string GetContainerName(string eventHubName, string eventHubNamespace, string storageConnectionString)
+        {
+            CloudStorageAccount account = CloudStorageAccount.Parse(storageConnectionString);
+            var client = account.CreateCloudTableClient();
```

Comment:
We do create table for the Functions (albeit not SDK) logging, and I followed that naming convention to try and be consistent.  (Although it'd be great to switch SDK over to that faster table logging). 

I preferred Table because it's better than Blob for small independent updates like this. For Blob, I didn't want a blob per entity (inefficient); and I didn't want a single blob that all entities shared (since the entities are independent from each other; I wanted to minimize concurrency cases) 

That said, if you'd rather this be a single blob in the  azure-webjobs-host ,  I can change it.